apps/backend/src/mcp/tools/system_monitoring.py

```
@@ -983,6 +962,286 @@ async def get_system_logs(
+        )
+        
+        return response
+        
+    except Exception as e:
+        logger.error(f"Error collecting drive stats from {device}: {e}")
+        raise SystemMonitoringError(
+            message=f"Failed to collect drive stats: {str(e)}",
+            device=device,
+            operation="get_drive_stats",
+            details={"error": str(e), "drive": drive}
+        )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)